bokehjs/src/lib/core/hittest.ts

```diff
@@ -24,6 +24,26 @@ export function point_in_poly(x: number, y: number, px: number[], py: number[]):
   return inside
 }
 
+export function point_in_ellipse(x: number, y: number, angle: number, b: number, a: number, x0: number, y0: number): boolean {
+  const A = ((Math.cos(angle) / a) ** 2 + (Math.sin(angle) / b) ** 2)
+  const B = 2 * Math.cos(angle) * Math.sin(angle) * ((1 / a) ** 2 - (1 / b) ** 2)
+  const C = ((Math.cos(angle) / b) ** 2 + (Math.sin(angle) / a) ** 2)
+  const eqn = A * (x - x0) ** 2 + B * (x - x0) * (y - y0) + C * (y - y0) ** 2
+  const inside = eqn <= 1
+  return inside
+}
+
+export function point_in_oval(x: number, y: number, angle: number, b: number, a: number, x0: number, y0: number): boolean {
+  // width of oval from cubic bezier = 3/4 * width ellipse
+  a = a * 0.75
+  const A = ((Math.cos(angle) / a) ** 2 + (Math.sin(angle) / b) ** 2)
+  const B = 2 * Math.cos(angle) * Math.sin(angle) * ((1 / a) ** 2 - (1 / b) ** 2)
+  const C = ((Math.cos(angle) / b) ** 2 + (Math.sin(angle) / a) ** 2)
+  const eqn = A * (x - x0) ** 2 + B * (x - x0) * (y - y0) + C * (y - y0) ** 2
+  const inside = eqn <= 1
+  return inside
```

Comment:
These two functions look almost identical, couldn't this just be:
```typescript
// width of oval from cubic bezier = 3/4 * width ellipse
export function point_in_oval(x: number, y: number, angle: number, b: number, a: number, x0: number, y0: number): boolean {
    return point_in_ellipse(x, y, angle, b, 0.75*x, x0, y0)
}

Reply:
Yes I agree it is largely redundant, left them seperate incase that was preferable for others.

Reply:
Less code to test and maintain is always preferable :)